code/modules/vehicles/wheelchair.dm

```diff
@@ -0,0 +1,118 @@
+/obj/vehicle/ridden/wheelchair
+	name = "wheelchair"
+	desc = "A chair with big wheels. It looks like you can move in this on your own."
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "wheelchair"
+	layer = OBJ_LAYER
+	max_integrity = 100
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 0, "bomb" = 10, "bio" = 0, "rad" = 0, "fire" = 20, "acid" = 30)
+	legs_required = 0
+	arms_required = 0
+	canmove = TRUE
+	density = FALSE
+	var/icon_overlay = "wheelchair_overlay"
+	var/list/drive_sounds = list('sound/effects/roll.ogg')
+	var/mob/living/carbon/human/H
+	var/mob/living/user
+	movedelay = 8
+
+/obj/vehicle/ridden/wheelchair/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 2.5
+	D.set_vehicle_dir_layer(SOUTH, OBJ_LAYER)
+	D.set_vehicle_dir_layer(NORTH, ABOVE_MOB_LAYER)
+	D.set_vehicle_dir_layer(EAST, OBJ_LAYER)
+	D.set_vehicle_dir_layer(WEST, OBJ_LAYER)
+
+/obj/vehicle/ridden/wheelchair/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/simple_rotation,ROTATION_ALTCLICK | ROTATION_CLOCKWISE, CALLBACK(src, .proc/can_user_rotate),CALLBACK(src, .proc/can_be_rotated),null)
+
+/obj/vehicle/ridden/wheelchair/Destroy()
+	if(obj_integrity <= 0)
+		new /obj/item/stack/rods(drop_location(), 1)
+		new /obj/item/stack/sheet/plasteel(drop_location(), 1)
+	if(has_buckled_mobs())
+		var/mob/living/carbon/H = buckled_mobs[1]
+		unbuckle_mob(H)
+	H = null
+	user = null
+	. = ..()
+
+/obj/vehicle/ridden/wheelchair/buckle_mob(mob/living/H, force = 0, check_loc = 1)
```

Comment:
Just replace this proc with Hippie's driver_move proc. This here is just spaghetti and doesn't take into account the fact that people can lose limbs while riding the wheelchair like Hippie does.